Mathlib/Order/SuccPred/CompleteLinearOrder.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2023 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.Order.SuccPred.Limit
+import Mathlib.Order.ConditionallyCompleteLattice.Basic
+
+/-!
+
+# Relation between `IsSuccLimit` and `iSup` in (conditionally) complete linear orders.
+
+-/
+
+open Order
+
+variable {ι α : Type*}
+
```

Comment:
IMHO, the basic lemma should be something like
```
lemma sSup_mem_of_not_isSuccLimit [ConditionallyCompleteLinearOrder α]
    {s : Set α} (hne : s.Nonempty) (hbdd : BddAbove s) (hlim : ¬ IsSuccLimit x (sSup s)) :
    sSup s ∈ s := _
```

or even
```
lemma CovBy.sSup_mem [ConditionallyCompleteLinearOrder α] {a : α}
    {s : Set α} (hcov : a ⋖ sSup s) (hne : s.Nonempty) (hbdd : BddAbove s) :
    sSup s ∈ s := _
```

Also, please discuss relation between primed and non-primed versions in the module docstring, in lemma docstrings, or in both places.